File: esykart-store/esykart-store-merchant/src/main/resources/public/app/components/category/js/categoryModel.js
function Category() {

	this.categoryId = null;
	this.name = null;
	this.longDescription = null;
	this.description = null;

	this.activeEndDate = null;
	this.activeStartDate = null;

	this.fulfillmentType = null;
	this.inventoryType = null;
	this.imageUrl = null;

	this.status = null;

	this.template = null;
	this.bannerId = null;
	this.url = null;
	this.parentCategory = new Object();

	this.categoryAttributes = new Array();

	this.setParentCategory = function(parentCategoryId) {
		var parentCategory = new Category();
		parentCategory.id = parentCategoryId;
		this.parentCategory = parentCategory;
	}
	
	this.addAttribute = function(id){
		var categoryAttribute = new CategoryAttribute();
		categoryAttribute.id = id;
		this.categoryAttributes.push(categoryAttribute);
	}

	this.toJson = function() {
		return JSON.stringify(this);
	}
	
	
}

File: esykart-apigateway/src/main/java/com/webientsoft/esykart/api/product/model/CategoryModel.java

package com.webientsoft.esykart.api.product.model;

import java.sql.Date;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonRootName;

@JsonRootName(value = "Category")
public class CategoryModel {

	private Integer categoryId;

	private Date activeEndDate;

	private Date activeStartDate;

	private Integer bannerId;

	private String description;

	private String fulfillmentType;

	private String imageUrl;

	private String inventoryType;

	private String longDescription;

	private String name;

	private String status;

	private String template;

	private String url;

	private List<CategoryAttributeModel> categoryAttributes;

	public Integer getCategoryId() {
		return categoryId;
	}

	public void setCategoryId(Integer categoryId) {
		this.categoryId = categoryId;
	}

	public Date getActiveEndDate() {
		return activeEndDate;
	}

	public void setActiveEndDate(Date activeEndDate) {
		this.activeEndDate = activeEndDate;
	}

	public Date getActiveStartDate() {
		return activeStartDate;
	}

	public void setActiveStartDate(Date activeStartDate) {
		this.activeStartDate = activeStartDate;
	}

	public Integer getBannerId() {
		return bannerId;
	}

	public void setBannerId(Integer bannerId) {
		this.bannerId = bannerId;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public String getFulfillmentType() {
		return fulfillmentType;
	}

	public void setFulfillmentType(String fulfillmentType) {
		this.fulfillmentType = fulfillmentType;
	}

	public String getImageUrl() {
		return imageUrl;
	}

	public void setImageUrl(String imageUrl) {
		this.imageUrl = imageUrl;
	}

	public String getInventoryType() {
		return inventoryType;
	}

	public void setInventoryType(String inventoryType) {
		this.inventoryType = inventoryType;
	}

	public String getLongDescription() {
		return longDescription;
	}

	public void setLongDescription(String longDescription) {
		this.longDescription = longDescription;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getStatus() {
		return status;
	}

	public void setStatus(String status) {
		this.status = status;
	}

	public String getTemplate() {
		return template;
	}

	public void setTemplate(String template) {
		this.template = template;
	}

	public String getUrl() {
		return url;
	}

	public void setUrl(String url) {
		this.url = url;
	}

	public List<CategoryAttributeModel> getCategoryAttributes() {
		return categoryAttributes;
	}

	public void setCategoryAttributes(List<CategoryAttributeModel> categoryAttributes) {
		this.categoryAttributes = categoryAttributes;
	}

}
